docs/reference/multi-tenancy.md

```diff
@@ -0,0 +1,383 @@
+
+## Multi-tenancy: Opt-In/Out
+
+There is no explicit feature flag to enable or disable multi-tenancy in Ratify. However, users would need to apply some resources to actively use multi-tenancy.
+
+### Custom Resources
+Users are supposed to use multi-tenancy by defining resources in a namespace scope or use cluster-wide resources as defaults. If users just want to enforce verification for the entire cluster, they should define resources in a cluster-wide scope.
+
+### Constraint Template
+Starting from version 1.2.0, Ratify now accommodates keys from external data request in both `[namespace]artifact-url` and `artifact-url` formats. To opt in multi-tenancy, users should adjust their constraint template to pass keys in `[namespace]artifact-url` format. However, if users just want to enforce verification for the entire cluster, they should pass only the `artifact-url` keys to Ratify.
+
+Examples:
+
+Constraint template using `artifact-url` formats: [default template](https://github.com/deislabs/ratify/blob/dev/library/default/template.yaml)
+
+Constraint template using `[namespace]artifact-url` formats: [multi-tenancy template](https://github.com/deislabs/ratify/blob/dev/library/multi-tenancy-validation/template.yaml)
+
+## Access Control
+As we [discussed](https://github.com/deislabs/ratify/blob/dev/docs/discussion/Multi-Tenancy%20Support.md#access-control) during multi-tenancy designing, Ratify will offload access control on namespaced level to Kubernetes cluster. Admins can use ABAC or RBAC authorization to control access to resources in a namespace. The restricted resources include Custom Resources in Ratify and general resources in the cluster.
+
+### User Group
+We defined 3 user groups in the multi-tenancy [design](https://github.com/deislabs/ratify/blob/dev/docs/discussion/Multi-Tenancy%20Support.md#user-group): cluster admin, team admin, and team developer. 
+1. Cluster admin has full control over the cluster and can manage all resources in the cluster. 
+2. Team admin can manage resources in a namespace and assign permissions to team developers. 
+3. Team developer can only access resources in a namespace and have limited permissions.
+
+Note: This is just a general example of user groups. Users can define more user groups based on their requirements.
+
+### Custom Resources
+Ratify provides a set of Custom Resources to define policies, verifiers, stores, and key management providers. Users can define these resources in namespaced level or cluster-wide scope. The access control to these resources is managed by the cluster admin. 
+
+For example, cluster admins are able to apply namespaced and cluster-wide CRs. Team admins have the capability to deploy namespaced CRs within their designated namespaces. Meanwhile, team devs are restricted to view CRs within their allocated namespaces.
```

Comment:
What is the resource description RBAC separation? Are team devs allowed to read resources from other namespaces?